internal/coreinternal/attraction/attraction.go

```diff
@@ -248,12 +276,27 @@ func (ap *AttrProc) Process(attrs pdata.AttributeMap) {
 	}
 }
 
-func getSourceAttributeValue(action attributeAction, attrs pdata.AttributeMap) (pdata.AttributeValue, bool) {
+func getAttributeValueFromContext(ctx context.Context, key string) (pdata.AttributeValue, bool) {
+	ci := client.FromContext(ctx)
+	vals := ci.Metadata.Get(key)
```

Comment:
Should this also support getting data from the auth object? Perhaps not as part of this PR though.

Reply:
It cold be easily done, in the future, if there is a use-case for that.